crates/test_probe/src/main.rs

```diff
@@ -0,0 +1,33 @@
+use composer::api::{Client, Event};
+use eyre::Result;
+use std::{env, thread, time::Duration};
+
+fn main() -> Result<()> {
+    color_eyre::install()?;
+
+    // Get server address, if given
+    let args: Vec<String> = env::args().collect();
+    let arg = args.get(1);
+    let server_address: Option<&str> = arg.map(|x: &String| x.as_str());
```

Comment:
Nit: I think we don't have to do the intermediate collect. Something like this should work (code untested):
```suggestion
    // Get server address, if given
    let aserver_address = env::args().next().map(|x| x.as_str());
```

Reply:
improved as part of d261261571191e188deed5b3c7ed99ab466a9e83